VerilogCodingStyle.md

```diff
@@ -2469,6 +2470,46 @@ module foo (
 endmodule
 ```
 
+#### Hierarchical references
+
+The use of hierarchical references in synthesizable RTL code is discouraged.
```

Comment:
We could even prohibit them. Do we have any examples where they're actually needed? If not all of our synthesis tooling supports them, I assume not?

Reply:
+1 to prohibit for synthesizable RTL.

We should definitely still allow them for assertions and such, these may be included in synthesizable RTL files but will be guarded by macros so they're stripped out when actually synthesizing.

Reply:
IIRC we have quite a few hierarchical references in the Ibex code base. However, they are just used to implement non-synthesizable features like RVFI and the trace module.

Reply:
Actually, also the RVFI stuff in Ibex is guarded by macros. So it would be fine to prohibit them with the exemption proposed by Greg. Let me change the PR accordingly.

Reply:
Alright, I've changed the wording accordingly, PTAL.